packages/python-sdk/e2b/sandbox_async/utils.py

```diff
@@ -1,7 +1,51 @@
-from typing import TypeVar, Union, Callable, Awaitable
+import httpx
+import logging
+
+from typing import TypeVar, Union, Callable, Awaitable, Optional
+
+from e2b.connection_config import ConnectionConfig
+from e2b.api import limits, AsyncApiClient
 
 T = TypeVar("T")
 OutputHandler = Union[
     Callable[[T], None],
     Callable[[T], Awaitable[None]],
 ]
+
+logger = logging.getLogger(__name__)
+
+
+def get_api_client(config: ConnectionConfig) -> AsyncApiClient:
+    return AsyncApiClient(
+        config,
+        transport=get_transport(config),
+    )
+
+
+class AsyncTransportWithLogger(httpx.AsyncHTTPTransport):
+    async def handle_async_request(self, request):
+        url = f"{request.url.scheme}://{request.url.host}{request.url.path}"
+        logger.info(f"Request: {request.method} {url}")
+        response = await super().handle_async_request(request)
+
+        # data = connect.GzipCompressor.decompress(response.read()).decode()
+        logger.info(f"Response: {response.status_code} {url}")
+
+        return response
+
+    @property
+    def pool(self):
+        return self._pool
+
+
+_transport: Optional[AsyncTransportWithLogger] = None
+
+
+def get_transport(config: ConnectionConfig) -> AsyncTransportWithLogger:
+    global _transport
```

Comment:
can we avoid globals?

Reply:
Yes, I think it would be better to create a class with static attribute and then export a shared instance of it.